PR_REVIEW_CHECKLIST.md

```diff
@@ -0,0 +1,98 @@
+3. Should we add regression testing for code introduced in this PR?
+    - If the PR introduces **new** pages, features, etc. then we **should** have new regression tests written. Please note this in your PR so that Applause can make sure it gets added.
+    - If the PR fixes a bug, Applause will check to make sure the existing regression testing steps already cover the workflow fixed by the bug.
+
+# Reviewing the code
+
+## Good code patterns to require
+
+1. Check that functions have comments when appropriate.
+    - If the function has params or returns something, use JSDocs syntax to describe them.
+        - Indicate the param name(s), data type passed and / or returned, and purpose if not immediately obvious.
+    - Obvious functions (with no params / return value) should not have comments.
+    - **In short: _Add comments & docs often, only when useful._**
+2. All copy / text shown to users in the product should be stored in the `src/languages/*` files for localization.
+3. Platform-specific files should follow the proper naming convention (see [Platform-Specific File Extensions](https://github.com/expensify/app#platform-specific-file-extensions))
+    - Example: `MyComponent/index.android.js`
+
+## Code patterns to check for improvement
+
+1. Is there appropriate error handling?
+2. Reuse...
+3. Are there any changes that could've been applied to one component that were applied to every usage of the component instead?
+    - Example: https://github.com/Expensify/App/pull/3129#pullrequestreview-668271068
+4. Ensure fixes for just android or ios aren't put in a `index.native.js`. Instead they should be in a specific `index.ios.js` or `index.android.js` file.
+    - Put another way, ensure platform-specific solutions are not applied cross-platform.
+5. Look for usages of react fragment wrappings and confirm they are necessary.
+    - Also look for any unnecessary `View`s and think if they should be replaced with React fragments.
+6. Were any new styles added? Can we reuse existing styles instead? Can we generalize the new style?
+
+## Code patterns to avoid
+
+1. Make sure any `setTimeout`s are absolutely necessary, not just a workaround for a situation that isn’t fully understood.
+2. Ensure Onyx is used appropriately. ([docs](https://github.com/expensify/react-native-onyx#merging-data))
+3. Don't mention the `isSmallScreenWidth` prop type explicitly in any component, use `...windowDimensionsPropTypes` instead.
+
+## Considerations around code reuse
+
+While trying to keep our code DRY (don't repeat yourself), you may find you're walking a thin line between refactoring & reusing a component vs. creating a brand new component. Here are some guidelines we use - they're not hard and fast rules, so please feel free to post in #expensify-open-source to gather peer opinions in order to make the best decisions.
+
+1. Specialization
```

Comment:
@Luke9389 I got this from your section "When to ask for help" - I decided to focus on code reuse ideas (specialize vs refactor) but feel free to suggest revisions :D

Reply:
I'd love for @marcaaron to check this out bc I was sort of adapting some of his comments from a few threads. I'm totally open to changes here. In general, I like the idea of an actual "When to Ask For Help" section that covers the stuff that doesn't neatly fit into checkboxes.

Reply:
This is great!

Reply:
@Luke9389 Do you think the "When to ask for help" can be super generic and can be used to help us improve the doc? Like generically "when to ask for help on a specific checklist item" or "when to ask for help on a specific PR b/c there's some considerations that aren't outlined in the doc yet"?

Reply:
I don't think I understand your question fully, but I'd prefer the section to cover stuff that's not on a checkbox. So more the latter.